From 01d20c92a98574e16a352bd276d170828288e0f7 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 05:17:54 +0200
Subject: [PATCH] [SW-58071] arch: create regmap and regmask files

This is preparatory patch for adding code for gaudi3. Currently
the common functions used by greco and gaudi2 are defined in
common file. Since the register and mask names for greco and gaudi2
were almost identical, there was not need to add regmap files for
registers used in common file. However in gaudi3, all registers
name and masks differ due to addition of die id. This patch creates
regmap and regmask files for such registers and masks for gaudi2 and
greco devices used in common function.

Change-Id: I3386c577ecaa6789c6a6fdf3223ce9100c6af0fa
---

diff --git a/arch/gaudi2/arc/regmap.h b/arch/gaudi2/arc/regmap.h
new file mode 100644
index 0000000..93143d2
--- /dev/null
+++ b/arch/gaudi2/arc/regmap.h
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GAUDI2_REGMAP_H_
+#define _GAUDI2_REGMAP_H_
+
+#define mmI2C_S_IC_CON			mmPSOC_I2C_S_IC_CON
+#define mmI2C_S_IC_SAR			mmPSOC_I2C_S_IC_SAR
+#define mmI2C_S_IC_INTR_MASK		mmPSOC_I2C_S_IC_INTR_MASK
+#define mmI2C_S_IC_RX_TL		mmPSOC_I2C_S_IC_RX_TL
+#define mmI2C_S_IC_TX_TL		mmPSOC_I2C_S_IC_TX_TL
+#define mmI2C_S_IC_ENABLE		mmPSOC_I2C_S_IC_ENABLE
+
+#define mmQSPI_SER			mmPSOC_QSPI_SER
+#define mmQSPI_BAUDR			mmPSOC_QSPI_BAUDR
+#define mmQSPI_DR0			mmPSOC_QSPI_DR0
+
+#define mmEFUSE_READ_CTRL		mmPSOC_EFUSE_READ_CTRL
+#define mmEFUSE_READ_ADDRESS		mmPSOC_EFUSE_READ_ADDRESS
+#define mmEFUSE_DATA_0			mmPSOC_EFUSE_DATA_0
+
+#define mmPRSTN				mmPSOC_GLOBAL_CONF_PRSTN
+#define mmCOLD_RST_FLOPS_2		mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2
+#define mmPCIE_PHY_RST_N		mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N
+#define mmBOOT_STRAP_PINS_L		mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L
+#define mmPRSTN_MASK			mmPSOC_GLOBAL_CONF_PRSTN_MASK
+#define mmSPI_DMA_AWUSER		mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER
+#define mmSPI_DMA_CTRL			mmPSOC_GLOBAL_CONF_SPI_DMA_CTRL
+#define mmSPI_DMA_STATUS		mmPSOC_GLOBAL_CONF_SPI_DMA_STATUS
+#define mmSPI_DMA_DST_ADDR_L		mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L
+#define mmSPI_DMA_DST_ADDR_H		mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_H
+#define mmMEM_CPY_CTRL			mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL
+#define mmMEM_CPY_STATUS		mmPSOC_GLOBAL_CONF_MEM_CPY_STATUS
+#define mmMEM_CPY_DEST_ADDR_H		mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H
+#define mmMEM_CPY_DEST_ADDR_L		mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L
+#define mmMEM_CPY_CTRL2			mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2
+#define mmMEM_CPY_AXUSER		mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER
+#define mmSPI_DIRECT_WR_RD_CTRL		mmPSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL
+
+#define mmGPIO0_DATA			mmPSOC_GPIO0_DATA
+
+#define mmSECURITY_MIN_FW_VERSION	mmPSOC_SECURITY_MIN_FW_VERSION
+#define mmSECURITY_PRI_IMG_STATUS	mmPSOC_SECURITY_PRI_IMG_STATUS
+#define mmSECURITY_SEC_IMG_STATUS	mmPSOC_SECURITY_SEC_IMG_STATUS
+#define mmSECURITY_PRI_IMG_FW_VER	mmPSOC_SECURITY_PRI_IMG_FW_VER
+#define mmSECURITY_SEC_IMG_FW_VER	mmPSOC_SECURITY_SEC_IMG_FW_VER
+#define mmSECURITY_STATUS		mmPSOC_SECURITY_STATUS
+#define mmSECURITY_KEY_REVOKE		mmPSOC_SECURITY_KEY_REVOKE
+#define mmSW_STATUS_0			mmPSOC_SECURITY_SW_STATUS_0
+
+#define mmJT_MSTR_IF_CORE_HBW_AXPROT_OVD	mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVD	mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVD	mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmCPU_HOT_RST			mmPCIE_WRAP_CPU_HOT_RST
+#define mmPCIE_RSP_OK			mmPCIE_WRAP_PCIE_RSP_OK
+
+#define mmLINK_CAPABILITIES_REG			mmPCIE_DBI_LINK_CAPABILITIES_REG
+#define mmLINK_CONTROL2_LINK_STATUS2_REG	mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG
+
+#define mmLTSSM_EN			mmPCIE_AUX_LTSSM_EN
+#define mmAPP_SRIS_MODE			mmPCIE_AUX_APP_SRIS_MODE
+
+#define mmPHY_EXT_CTRL_SEL		mmPCIE_PHY_PHY_EXT_CTRL_SEL
+#define mmPIPE_RX_SRIS_MODE_EN		mmPCIE_PHY_PIPE_RX_SRIS_MODE_EN
+#define mmEXT_MPLLA_SSC_PEAK		mmPCIE_PHY_EXT_MPLLA_SSC_PEAK
+#define mmEXT_MPLLA_SSC_STEP_SIZE	mmPCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE
+#define mmEXT_MPLLB_SSC_PEAK		mmPCIE_PHY_EXT_MPLLB_SSC_PEAK
+#define mmEXT_MPLLB_SSC_STEP_SIZE	mmPCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE
+
+#endif /* _GAUDI2_REGMAP_H_ */
diff --git a/arch/gaudi2/arc/regmask.h b/arch/gaudi2/arc/regmask.h
new file mode 100644
index 0000000..2769cb2
--- /dev/null
+++ b/arch/gaudi2/arc/regmask.h
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GAUDI2_REGMASK_H_
+#define _GAUDI2_REGMASK_H_
+
+#define SPI_DMA_CTRL_ADDR_SHIFT			PSOC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT
+#define SPI_DMA_STATUS_ERROR_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_MASK
+#define SPI_DMA_STATUS_ERROR_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_SHIFT
+#define SPI_DMA_STATUS_DONE_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_MASK
+#define SPI_DMA_CTRL_FLD_SIZE_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_SHIFT
+#define SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK	\
+				PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK
+#define PRSTN_MASK_IND_MASK			PSOC_GLOBAL_CONF_PRSTN_MASK_IND_MASK
+#define MEM_CPY_CTRL2_USE_CONST_SHIFT		PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_SHIFT
+
+#define SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT	\
+				PSOC_SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT
+#define SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT	\
+				PSOC_SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT
+#define SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT	PSOC_SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT
+#define SECURITY_STATUS_IMG_VALIDATED_SHIFT	PSOC_SECURITY_STATUS_IMG_VALIDATED_SHIFT
+#define SECURITY_STATUS_FW_UPGRADE_SHIFT	PSOC_SECURITY_STATUS_FW_UPGRADE_SHIFT
+#define SECURITY_STATUS_KEY_REVOKED_SHIFT	PSOC_SECURITY_STATUS_KEY_REVOKED_SHIFT
+#define SECURITY_STATUS_KEY_REVOKED_IND_SHIFT	PSOC_SECURITY_STATUS_KEY_REVOKED_IND_SHIFT
+#define SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT	PSOC_SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT
+#define SECURITY_STATUS_FW_VER_INC_SHIFT	PSOC_SECURITY_STATUS_FW_VER_INC_SHIFT
+#define SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT	PSOC_SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT
+#define SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT	\
+					PSOC_SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT
+
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK	\
+				PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT	\
+				PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK	\
+		PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT	\
+		PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT
+
+#define LTSSM_EN_IND_MASK			PCIE_AUX_LTSSM_EN_IND_MASK
+
+#define CPU_HOT_RST_CTRL_MASK			PCIE_WRAP_CPU_HOT_RST_CTRL_MASK
+
+#endif /* _GAUDI2_REGMASK_H_ */
diff --git a/arch/greco/arc_elf32/regmap.h b/arch/greco/arc_elf32/regmap.h
new file mode 100644
index 0000000..e3a6653
--- /dev/null
+++ b/arch/greco/arc_elf32/regmap.h
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GRECO_REGMAP_H_
+#define _GRECO_REGMAP_H_
+
+#define mmI2C_S_IC_CON			mmPSOC_I2C_S_IC_CON
+#define mmI2C_S_IC_SAR			mmPSOC_I2C_S_IC_SAR
+#define mmI2C_S_IC_INTR_MASK		mmPSOC_I2C_S_IC_INTR_MASK
+#define mmI2C_S_IC_RX_TL		mmPSOC_I2C_S_IC_RX_TL
+#define mmI2C_S_IC_TX_TL		mmPSOC_I2C_S_IC_TX_TL
+#define mmI2C_S_IC_ENABLE		mmPSOC_I2C_S_IC_ENABLE
+
+#define mmQSPI_SER			mmPSOC_QSPI_SER
+#define mmQSPI_BAUDR			mmPSOC_QSPI_BAUDR
+#define mmQSPI_DR0			mmPSOC_QSPI_DR0
+
+#define mmEFUSE_READ_CTRL		mmPSOC_EFUSE_READ
+#define mmEFUSE_READ_ADDRESS		mmPSOC_EFUSE_READ_ADDRESS
+#define mmEFUSE_DATA_0			mmPSOC_EFUSE_DATA_0
+
+#define mmPRSTN				mmPSOC_GLOBAL_CONF_PRSTN
+#define mmCOLD_RST_FLOPS_2		mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2
+#define mmPCIE_PHY_RST_N		mmPSOC_GLOBAL_CONF_PCIE_PHY_RST_N
+#define mmBOOT_STRAP_PINS_L		mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L
+#define mmPRSTN_MASK			mmPSOC_GLOBAL_CONF_PRSTN_MASK
+#define mmSPI_DMA_AWUSER		mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER
+#define mmSPI_DMA_CTRL			mmPSOC_GLOBAL_CONF_SPI_DMA_CTRL
+#define mmSPI_DMA_STATUS		mmPSOC_GLOBAL_CONF_SPI_DMA_STATUS
+#define mmSPI_DMA_DST_ADDR_L		mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L
+#define mmSPI_DMA_DST_ADDR_H		mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_H
+#define mmMEM_CPY_CTRL			mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL
+#define mmMEM_CPY_STATUS		mmPSOC_GLOBAL_CONF_MEM_CPY_STATUS
+#define mmMEM_CPY_DEST_ADDR_H		mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H
+#define mmMEM_CPY_DEST_ADDR_L		mmPSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L
+#define mmMEM_CPY_CTRL2			mmPSOC_GLOBAL_CONF_MEM_CPY_CTRL2
+#define mmMEM_CPY_AXUSER		mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER
+#define mmSPI_DIRECT_WR_RD_CTRL		mmPSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL
+
+#define mmGPIO0_DATA			mmPSOC_GPIO0_DATA
+
+#define mmSECURITY_MIN_FW_VERSION	mmPSOC_SECURITY_MIN_FW_VERSION
+#define mmSECURITY_PRI_IMG_STATUS	mmPSOC_SECURITY_PRI_IMG_STATUS
+#define mmSECURITY_SEC_IMG_STATUS	mmPSOC_SECURITY_SEC_IMG_STATUS
+#define mmSECURITY_PRI_IMG_FW_VER	mmPSOC_SECURITY_PRI_IMG_FW_VER
+#define mmSECURITY_SEC_IMG_FW_VER	mmPSOC_SECURITY_SEC_IMG_FW_VER
+#define mmSECURITY_STATUS		mmPSOC_SECURITY_STATUS
+#define mmSECURITY_KEY_REVOKE		mmPSOC_SECURITY_KEY_REVOKE
+#define mmSW_STATUS_0			mmPSOC_SECURITY_SW_STATUS_0
+
+#define mmJT_MSTR_IF_CORE_HBW_AXPROT_OVD	mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVD	mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVD	mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD
+
+#define mmCPU_HOT_RST			mmPCIE_WRAP_CPU_HOT_RST
+#define mmPCIE_RSP_OK			mmPCIE_WRAP_PCIE_RSP_OK
+
+#define mmLINK_CAPABILITIES_REG			mmPCIE_DBI_LINK_CAPABILITIES_REG
+#define mmLINK_CONTROL2_LINK_STATUS2_REG	mmPCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG
+
+#define mmLTSSM_EN			mmPCIE_AUX_LTSSM_EN
+#define mmAPP_SRIS_MODE			mmPCIE_AUX_APP_SRIS_MODE
+
+#define mmPHY_EXT_CTRL_SEL		mmPCIE_PHY_PHY_EXT_CTRL_SEL
+#define mmPIPE_RX_SRIS_MODE_EN		mmPCIE_PHY_PIPE_RX_SRIS_MODE_EN
+#define mmEXT_MPLLA_SSC_PEAK		mmPCIE_PHY_EXT_MPLLA_SSC_PEAK
+#define mmEXT_MPLLA_SSC_STEP_SIZE	mmPCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE
+#define mmEXT_MPLLB_SSC_PEAK		mmPCIE_PHY_EXT_MPLLB_SSC_PEAK
+#define mmEXT_MPLLB_SSC_STEP_SIZE	mmPCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE
+
+#endif /* _GRECO_REGMAP_H_ */
diff --git a/arch/greco/arc_elf32/regmask.h b/arch/greco/arc_elf32/regmask.h
new file mode 100644
index 0000000..9fde615
--- /dev/null
+++ b/arch/greco/arc_elf32/regmask.h
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GRECO_REGMASK_H_
+#define _GRECO_REGMASK_H_
+
+#define SPI_DMA_CTRL_ADDR_SHIFT			PSOC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT
+#define SPI_DMA_STATUS_ERROR_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_MASK
+#define SPI_DMA_STATUS_ERROR_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_SHIFT
+#define SPI_DMA_STATUS_DONE_MASK		PSOC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_MASK
+#define SPI_DMA_CTRL_FLD_SIZE_SHIFT		PSOC_GLOBAL_CONF_SPI_DMA_CTRL_FLD_SIZE_SHIFT
+#define SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK	\
+				PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK
+#define PRSTN_MASK_IND_MASK			PSOC_GLOBAL_CONF_PRSTN_MASK_IND_MASK
+#define MEM_CPY_CTRL2_USE_CONST_SHIFT		PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_SHIFT
+
+#define SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT	\
+				PSOC_SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT
+#define SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT	\
+				PSOC_SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT
+#define SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT	PSOC_SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT
+#define SECURITY_STATUS_IMG_VALIDATED_SHIFT	PSOC_SECURITY_STATUS_IMG_VALIDATED_SHIFT
+#define SECURITY_STATUS_FW_UPGRADE_SHIFT	PSOC_SECURITY_STATUS_FW_UPGRADE_SHIFT
+#define SECURITY_STATUS_KEY_REVOKED_SHIFT	PSOC_SECURITY_STATUS_KEY_REVOKED_SHIFT
+#define SECURITY_STATUS_KEY_REVOKED_IND_SHIFT	PSOC_SECURITY_STATUS_KEY_REVOKED_IND_SHIFT
+#define SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT	PSOC_SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT
+#define SECURITY_STATUS_FW_VER_INC_SHIFT	PSOC_SECURITY_STATUS_FW_VER_INC_SHIFT
+#define SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT	PSOC_SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT
+#define SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT	\
+					PSOC_SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT
+
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK	\
+				PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT	\
+				PCIE_DBI_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK	\
+		PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT	\
+		PCIE_DBI_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT
+
+#define LTSSM_EN_IND_MASK			PCIE_AUX_LTSSM_EN_IND_MASK
+
+#define CPU_HOT_RST_CTRL_MASK			PCIE_WRAP_CPU_HOT_RST_CTRL_MASK
+
+#endif /* _GRECO_REGMASK_H_ */
